app/Http/Controllers/JobsController.php

```diff
@@ -277,10 +301,12 @@ public function action(Request $request)
     {
 
         $action = $request->input('action');
+
+
         switch ($action) {
             case 'save': 
                 $result = $this->save($request->all());
-                if ($result['status']) {
+                if ($result['success']) {
```

Comment:
Để tránh case nào cũng lặp lại đoạn này:
```
// ...
else { 
    return redirect()->route('jobs.create')->withInput()->withErrors($result['message']);
}

```
Nên viết lại theo dạng try/catch

```
try {
    switch ($action) {
        case 'save':
            $result = $this->save($request->all());
         case 'save_copy':
            // ...
    }
} catch (Throwable $message) {
    return redirect()->route('jobs.create')->withInput()->withErrors($message);
}
```
(Các logic viết trong `$this->save` cũng không` return $result['error']` mà `throw  $result['error']` )